lib/ansible/modules/system/selinux.py

```diff
@@ -195,26 +226,30 @@ def main():
                 if runtime_state != 'permissive':
                     # Temporarily set state to permissive
                     set_state(module, 'permissive')
-                    msgs.append('runtime state temporarily changed from \'%s\' to \'permissive\', state change will take effect next reboot' % (runtime_state))
+                    module.warn('SELinux state temporarily changed from \'%s\' to \'permissive\'. State change will take effect next reboot.' % (runtime_state))
                 else:
-                    msgs.append('state change will take effect next reboot')
+                    module.warn('SELinux state change will take effect next reboot')
+                reboot_required = True
             else:
                 set_state(module, state)
-                msgs.append('runtime state changed from \'%s\' to \'%s\'' % (runtime_state, state))
+                msgs.append('SELinux state changed from \'%s\' to \'%s\'' % (runtime_state, state))
+
+                # Only report changes if the file is changed.
+                # This prevents the task from reporting changes every time the task is run.
+                changed = True
```

Comment:
Putting the `changed = True` here means that changes will not be reported every time the module is run. Instead, I set `reboot_required` and return that as an indicator that the system needs a reboot. This may not be desirable and I'm open to suggestions.

Reply:
Seems reasonable. It is always better to have info this (a flag to indicate the system needs reboot) as predictable structured data so that automation can act on it reliably. Previously this would have to have been parsed out of the msgs (or more likely, the msg was rarely noticed...) so that is an improvement.

If someone was parsing the message, the msg change might break it, but structured data and a better error message is 👍 for me